# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: 09db50c8f370
# Full Hash: 09db50c8f370b69f186f3e6b7ce6154a984b40b4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Pass codec configs by value/reference instead of pointers. r=ng
#   
#   This patch mainly sets the stage for using codec configs with StateMirroring.
#   It also makes the code a bit simpler, and removes the need to handle the nullptr
#   case.
# ==============================================================================

diff -r d1552c620c11 -r 09db50c8f370 dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Wed Aug 11 11:33:21 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Wed Apr 14 17:36:40 2021 +0200
@@ -67,7 +67,7 @@
    *       restarting transmission sub-system on the engine.
    */
   MediaConduitErrorCode ConfigureSendMediaCodec(
-      const AudioCodecConfig* codecConfig) override;
+      const AudioCodecConfig& codecConfig) override;
   /**
    * Function to configure list of receive codecs for the audio session
    * @param sendSessionConfig: CodecConfiguration
@@ -79,7 +79,7 @@
    *       restarting transmission sub-system on the engine.
    */
   MediaConduitErrorCode ConfigureRecvMediaCodecs(
-      const std::vector<UniquePtr<AudioCodecConfig>>& codecConfigList) override;
+      const std::vector<AudioCodecConfig>& codecConfigList) override;
 
   MediaConduitErrorCode SetLocalRTPExtensions(
       MediaSessionConduitLocalDirection aDirection,
@@ -180,14 +180,14 @@
   void operator=(const WebrtcAudioConduit& other) = delete;
 
   // Function to convert between WebRTC and Conduit codec structures
-  bool CodecConfigToWebRTCCodec(const AudioCodecConfig* codecInfo,
+  bool CodecConfigToWebRTCCodec(const AudioCodecConfig& codecInfo,
                                 webrtc::AudioSendStream::Config& config);
 
   // Generate block size in sample length for a given sampling frequency
   unsigned int GetNum10msSamplesForFrequency(int samplingFreqHz) const;
 
   // Checks the codec to be applied
-  MediaConduitErrorCode ValidateCodecConfig(const AudioCodecConfig* codecInfo,
+  MediaConduitErrorCode ValidateCodecConfig(const AudioCodecConfig& codecInfo,
                                             bool send);
 
   MediaConduitErrorCode CreateSendStream();